pkg/apis/pipeline/v1beta1/resolver_types.go

```diff
@@ -33,16 +33,5 @@ type ResolverRef struct {
 	// "repo" or "path" but the set of params ultimately depends on
 	// the chosen resolver.
 	// +optional
-	// +listType=atomic
-	Resource []ResolverParam `json:"resource,omitempty"`
-}
-
-// ResolverParam is a single parameter passed to a resolver.
-type ResolverParam struct {
-	// Name is the name of the parameter that will be passed to the
-	// resolver.
-	Name string `json:"name"`
-	// Value is the string value of the parameter that will be
-	// passed to the resolver.
-	Value string `json:"value"`
+	Resource map[string]string `json:"resource,omitempty"`
```

Comment:
Very good question! I am not a fan of `resource` here - it's not really clear what it means in the first place. I feel like just `params` would work, though since `params` is used in a bunch of places with different syntax than here, I'm not sure if that's ideal...

Reply:
In the TEP, we used `resource` ? I'd rather go with `params` or `values` than `resource` indeed.